Topics/Tech_Stacks/Tiptap/Tiptap.md

```diff
@@ -1,2 +1,284 @@
+# A Guide on using Tiptap in a React CRUD app.
 
-Test
+## What Is Tiptap
+
+Tiptap, one of the leading open-source dev tools for rich content editors is trusted by many companies like Substack, AxiosHQ, and GitLab.[^1]
+
+They have accumulated over 2 million downloads[^1] with their cool products such as Tiptap Content AI and Tiptap Collobatoration. 
+
+
+![Untitled design (4)](https://github.com/AshkanAleshams/learning-software-engineering.github.io/assets/90326959/f56018c2-e7ef-478f-95d2-de19838cc1c7)
+
+
+However for this guide, we will focus on the Tiptap Editor.
+
+Tiptap Editor is a headless toolkit for building rich text WYSIWYG editors for Vue.js and React.js applications. Giving the user the complete freedom to fashion their own editor interface using customizable building blocks. 
+It offers sensible defaults, an extensive array of extensions, and an intuitive API for tailoring every facet to the user's preferences.
+
+
+
+## Why Use Tiptap
+
+If you are looking for a powerful text editor for web applications, Tiptap is a great option to consider. Being open-source and built on the well-regarded ProseMirror library,[^3] Tiptap offers a solid foundation. It features real-time collaboration and boasts a rich library of [community extensions](https://github.com/ueberdosis/awesome-tiptap#community-extensions) that includes an extension that enables Markdown support. Starting with Tiptap is easy thanks to its free tier, but for those needing advanced features like AI integration and document commenting, [paid plans](https://tiptap.dev/pricing) are available, making Tiptap a versatile solution for diverse project needs.
+
+## Relevant Links
+_Relevant links such s Tiptap's documentation page, etc_
+
+## Requirements
+
+There are multiple methods to install Tiptap, each tailored to a specific use case. Detailed integration guides for different project types can be accessed [here](https://tiptap.dev/docs/editor/installation). In this article, we'll focus on creating a React project integrated with Tiptap, so you'll need Node.js and npm; if you don't have them installed, follow [this guide](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) to install them.
+
+## Intro to Tiptap in React
+
+_Include gifs, code snippets, explainations_
+
+**1. Create a Project** 
+
+You may use any other method of creating a React app if you wish, but 
+[Tiptap Create React App template by @alb](https://github.com/alb/cra-template-tiptap) is the quickest way. 
+
+``` 
+npx create-react-app my-tiptap-project --template tiptap
+
+```
+
+This will also install the dependencies for you!
+
+
+**2. Run the Project**
+
+Start your React application to see TipTap in action.
+```
+cd my-tiptap-project
+
+npm start
+```
+
+**3. Add a Menu Bar:**
+
+Improve your editor by adding a menu bar to Tiptap.jsx.
+
+
+```
```

Comment:
The length of the code might overwhelm the reader, so putting the code inside a collapsible details tag could make it easier to read:
````
<details>
  <summary>Tiptap.jsx</summary>
  ```
  [code]
  ```
</details>
````